packages/playwright-core/src/server/recorder/java.ts

```diff
@@ -26,19 +26,37 @@ import { JavaScriptFormatter } from './javascript';
 import { escapeWithQuotes } from '../../utils/isomorphic/stringUtils';
 import { asLocator } from '../../utils/isomorphic/locatorGenerators';
 
+type JavaLanguageMode = 'library' | 'junit';
+
 export class JavaLanguageGenerator implements LanguageGenerator {
-  id = 'java';
+  id: string;
   groupName = 'Java';
-  name = 'Library';
+  name: string;
   highlighter = 'java' as Language;
+  _mode: JavaLanguageMode;
+
+  constructor(mode: JavaLanguageMode) {
+    if (mode === 'library') {
+      this.name = 'Library';
+      this.id = 'java';
+    } else if (mode === 'junit') {
+      this.name = 'JUnit';
+      this.id = 'java-junit';
+    } else {
+      throw new Error(`Unknown Java language mode: ${mode}`);
+    }
+    this._mode = mode;
+  }
 
   generateAction(actionInContext: ActionInContext): string {
     const action = actionInContext.action;
     const pageAlias = actionInContext.frame.pageAlias;
-    const formatter = new JavaScriptFormatter(6);
+    const offset = this._mode === 'junit' ? 4 : 6;
+    const formatter = new JavaScriptFormatter(offset);
 
     if (action.name === 'openPage') {
-      formatter.add(`Page ${pageAlias} = context.newPage();`);
+      if (this._mode !== 'junit' || pageAlias !== 'page')
+        formatter.add(`Page ${pageAlias} = context.newPage();`);
```

Comment:
In this case we also need to add `BrowserContext context` parameter to the test function.

Reply:
After looking at other generators, I did the same thing in the case it is a test : nothing.  
Means `openpage` and and `closepage` are not managed, in the same way as tests in other languages.

We can improve this in another issue/PR. WDYT ?